plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueConverter.java

```diff
@@ -176,7 +176,12 @@ else if (isIcebergTable(tableParameters)) {
         }
         else {
             boolean isCsv = sd.serdeInfo() != null && HiveStorageFormat.CSV.getSerde().equals(sd.serdeInfo().serializationLibrary());
-            dataColumns = fromGlueColumns(sd.columns(), ColumnType.DATA, isCsv);
+            if (isDeltaLakeTable(tableParameters)) {
```

Comment:
Any reason not to update the above condition handling Iceberg format at L166? 
```java
        else if (isIcebergTable(tableParameters)) {
```

Reply:
Are you asking if the condition on `isDeltaLakeTable(tableParameters)` can be moved one lever higher, or can it be added to the condition for Iceberg? I'm not sure where `storage` is used, so I wanted to preserve how it's built. `dataColumns` for Iceberg is also built differently. I can move the condition one level higher, but that'll duplicate some code. LMK if that's ok.

Reply:
I meant adding to the Iceberg condition, but a separate condition seems better. Can we change like this? 
```java
        else if (isIcebergTable(tableParameters)) {
        ...
        }
        else if (isDeltaLakeTable(tableParameters)) {
            dataColumns = ImmutableList.of(FAKE_COLUMN);
            partitionColumns = ImmutableList.of();
            storage = fromGlueStorage(sd, databaseName + "." + glueTable.name());
        }
```

By the way, I think the existing logic still fails when `timestamp_ntz` type is used in partition columns. Could you add such a partition column in a test?

Reply:
Much better, and great catch, thanks!